frontend/src/components/pages/topics/Tab.Messages/index.tsx

```diff
@@ -65,6 +65,51 @@ interface TopicMessageViewProps {
         - add back summary of quick search  <this.FilterSummary />
 */
 
+const getStringValue = (value: string | TopicMessage): string => typeof value === 'string' ? value : JSON.stringify(value, null, 4)
+
+const CopyDropdown: FC<{ record: TopicMessage, onSaveToFile: Function }> = ({record, onSaveToFile}) => {
+
+    const toast = useToast()
+    return (
+        <Menu>
+            <Menu.Item key="0" disabled={record.key.isPayloadNull} onClick={() => {
+                navigator.clipboard.writeText(getStringValue(record)).then(() => {
+                    toast({
+                        status: 'success',
+                        description: 'Key copied to clipboard'
+                    })
+                })
+            }}>
+                Copy Key
+            </Menu.Item>
+            <Menu.Item key="2" disabled={record.value.isPayloadNull} onClick={() => {
```

Comment:
Why not key 1, 2, 3, 4?

Reply:
@malinskibeniamin this was there from before, I can surely refactor all of this, the question is to what level should I dive when refactoring :) Right now I do only what is necessary to migrate away from Antd and inline styles, but if you want I can go deeper by default.

Reply:
No worries, just asking - please feel free to leave it for now.